tests/test_triton_client.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import unittest
+import sys
+
+from model_analyzer.triton.server import TritonServerFactory, TritonServerConfig
+from model_analyzer.triton.client import TritonClientFactory, TritonClientConfig
+
+# Test parameters
+MODEL_LOCAL_PATH = '/model_analyzer/models'
+MODEL_REPOSITORY_PATH = '/model_analyzer/models'
+TRITON_VERSION = '20.09'
+CONFIG_TEST_ARG = 'url'
+CLIENT_TEST_PROTOCOLS = [('http', 'localhost:8000'),('grpc', 'localhost:8001')]
+TEST_MODEL_NAME = 'classification_chestxray_v1'
+
+class TestTritonClientMethods(unittest.TestCase):
+    
+    def setUp(self):
+        
+        # Create server config
+        self.server_config = TritonServerConfig()
+        self.server_config['model-repository'] = MODEL_REPOSITORY_PATH
+        self.server_config['model-control-mode'] = 'explicit'
+
+        # Set CUDA_VISIBLE_DEVICES
+        os.environ['CUDA_VISIBLE_DEVICES'] = '0'
+
+        # Create and start the server
+        self.server_factory = TritonServerFactory()
+        try:    
```

Comment:
Is there a way to wrap this call in a `self.assert_xyz(...)` test?

Reply:
I don't understand. Am I trying to ensure that self.server_factory is defined? Maybe I could try `self.assertIsInstance(self.server_factory, TritonServerFactory)`

Reply:
So in C++ Boost:test, we have a test for something like `test_no_throw(obj = do_something())`. Basically, it will fail if any exception is thrown. This way, you don't have to wrap the call in a `try...catch` statement.

Reply:
I think `unittest` has an `assertRaises()` but nothing for the opposite. In my latest commit I've just removed the try catch, as the test will fail if an exception is thrown anyway.